docs/signing-spec.md

```diff
@@ -342,6 +343,236 @@ For example, if a package is signed by an unknown host key, then this package ha
+
+To verify a `yanked_signature`, an agent:
+
+- MUST verify that the host key used to sign is in the agent's keyring
+    - An agent MAY refuse to treat a bindle as yanked if the key is not known
+    - An agent SHOULD notify the user if a bindle is yanked, but the yanking host key is not known
+- MUST verify that the signature is valid (using the same formula described for regular signature validation).
+- SHOULD verify that the timestamp for yanking is not the same as the signature block's `at` timestamp.
+    - This is a trivial protection against attempts to forge a yank block.
+- SHOULD require that the host key used to yank is also the same host key originally used to sign
+    - This is complicated by host key rotation. Our recommendation is that on host key rotation, all packages should be re-signed using the new host key so that it is always clear who the host authority is.
+
+
+The format of the `yanked_signature` is the same as the format of a regular signature:
+
+```toml
+bindle_version = "1.0.0"
+yanked = true
+
+[[yanked_signature]]
+by = "Matt Butcher <matt.butcher@example.com>"
+signature = "ddd237895ac..."
+key = "1c44..."
+role = "host"
+at = 1611960337
+```
+
+Note that this strategy does not prevent a key from being used to generate a known-bad signature so that the validation fails.
+
+### Known Issues with Yank Signing
+
+The current model does not provide a way to say "this package should never be trusted again" except for generating known-bad signatures.
+
+Likewise, while signing with `host` keys indicates that the host has yanked the package, there is no clear way to indicate the creator's intent, or a verifier's stance. The reason for this is that yanking is a host-specific operation, and does not have
+to do directly with creator intent. (In other words, the creator is not the authority on whether a bindle is yanked. The host is.)
+
+A compromised host could yank all packages, which may have dire initial consequences.
+However, these consequences are probably the correct consequences when a compromise happens, as the result is a denial of service from the host to the agent.
+
+### Possible Alternatives to Yank Signing
+
+The `yanked` field could be included in the main signing metadata.
+This would create a _strong yank_ under which the yank would invalidate all signatures.
+
+The main flaw with this route is that it makes a strong claim about what `yanked` means.
+In this case, `yanked` means "it is never secure to use this package", which may be true in some cases, but not all.
+
+Another issue with this approach is that a yanked bindle's non-host signatures would all be initially invalidated by the
+yank operation, even though the actual content of the package has not changed.
+This would require verifiers and creators to go _re-sign_ packages again after the `yanked` field was toggled to true.
```

Comment:
This could also lead to all sorts of other problems too right? Someone else could claim to be a creator when they aren't. I actually think we may want to specify that signatures are append only

Reply:
I think `creator` is a one-time thing, and I think we should honestly enforce that restriction when the API accepts the invoice.

The only case I can think of where an `approver` would want to modify is if the `approver` discovered a package was not to be trusted, but could not cause the `host` to yank the bindle.

I am less certain about proxies and hosts, where key rotation is desirable. During key rotation, I _think_ that we would want to remove the old key because it is no longer to be trusted. But maybe there would actually be no harm in appending a new signature and not removing the old one.

Reply:
Updated that signatures are append-only. This sounds like the right default option. We can relax iff necessary.